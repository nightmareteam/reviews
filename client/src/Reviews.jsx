import React from 'react';
import Review from './Review.jsx';
//import styled from 'styled-components';
const styled = window.styled;

/**
 * The reviews component displays all reviews from an array
 * @param {Array} reviews - an array of review objects
 * @param {String} sort - a string indicating the sort order of reviews 
 */
const Reviews = ({reviews, sort}) => (
  <ReviewsWrapper> 
    <Title>MOST {sort.toUpperCase()} REVIEWS</Title>
    <Subtitle>IN THE PAST 30 DAYS</Subtitle>
    {
      reviews.map(review => <Review key={review.post_id} review={review} source={'standard'}/>)
    }
  </ReviewsWrapper>
);

const Title = styled.div`
  padding-top: 20px;
  font-size: 14px;
  padding-right: 5px;
  color: white;
  display: inline;
`;

const Subtitle = styled(Title)`
  color: #56707f;
`;

const ReviewsWrapper = styled.div`
  float: left;
  width: 100%;
  font-family: "Motiva Sans", Arial, Helvetica, sans-serif;
  background: inherit;
  @media only screen and (min-width: 768px) {
    width: auto;
  }
`;

export default Reviews;