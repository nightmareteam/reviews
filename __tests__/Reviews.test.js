/* eslint-disable camelcase */
import React from 'react';
import { shallow, mount } from 'enzyme';
import Reviews from '../client/src/Reviews.jsx';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';
import $ from 'jquery';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
let props;
let mockCallback;

const createProps = (mockCallback) => ({
  reviews: [{
    content: 'Voluptatem voluptatem officia et.',
    createdAt: '2019-05-20T01:30:23.000Z',
    helpful_funny_count: 309,
    helpful_no_count: 503,
    helpful_yes_count: 993,
    hours_played: '2919',
    language: 'Turkish',
    post_id: 89613,
    product_count: 77,
    recommended: false,
    registration_date: '2018-11-24',
    review_count: 33,
    review_date: '2019-01-20',
    steam_level: 28,
    updatedAt: '2019-05-20T01:30:23.000Z',
    user_avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/blakesimkins/128.jpg',
    user_id: 88438,
    username: 'Salvador.OKon17'
  },
  {
    content: 'Voluptatem voluptatem officia et.',
    createdAt: '2019-05-20T01:30:23.000Z',
    helpful_funny_count: 39,
    helpful_no_count: 53,
    helpful_yes_count: 93,
    hours_played: '219',
    language: 'Spanish',
    post_id: 8913,
    product_count: 77,
    recommended: false,
    registration_date: '2018-11-23',
    review_count: 33,
    review_date: '2019-01-21',
    steam_level: 28,
    updatedAt: '2019-05-20T01:30:23.000Z',
    user_avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/blakesimkins/128.jpg',
    user_id: 88435,
    username: 'Salvador'
  }],
  sort: 'helpful'
});

beforeEach(() => {
  jest.resetModules();
  mockCallback = jest.fn();
  props = createProps(mockCallback);
  wrapper = shallow(<Reviews {...props} /> );
});

describe('Reviews component', () => {
  it('should render elements', () => {
    expect(wrapper).toMatchSnapshot();
  });

});

