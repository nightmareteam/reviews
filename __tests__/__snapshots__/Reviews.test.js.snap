// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews component should render elements 1`] = `
<styled.div>
  <styled.div>
    MOST 
    HELPFUL
     REVIEWS
  </styled.div>
  <Styled(styled.div)>
    IN THE PAST 30 DAYS
  </Styled(styled.div)>
  <Review
    key="89613"
    review={
      Object {
        "content": "Voluptatem voluptatem officia et.",
        "createdAt": "2019-05-20T01:30:23.000Z",
        "helpful_funny_count": 309,
        "helpful_no_count": 503,
        "helpful_yes_count": 993,
        "hours_played": "2919",
        "language": "Turkish",
        "post_id": 89613,
        "product_count": 77,
        "recommended": false,
        "registration_date": "2018-11-24",
        "review_count": 33,
        "review_date": "2019-01-20",
        "steam_level": 28,
        "updatedAt": "2019-05-20T01:30:23.000Z",
        "user_avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/blakesimkins/128.jpg",
        "user_id": 88438,
        "username": "Salvador.OKon17",
      }
    }
    source="standard"
  />
  <Review
    key="8913"
    review={
      Object {
        "content": "Voluptatem voluptatem officia et.",
        "createdAt": "2019-05-20T01:30:23.000Z",
        "helpful_funny_count": 39,
        "helpful_no_count": 53,
        "helpful_yes_count": 93,
        "hours_played": "219",
        "language": "Spanish",
        "post_id": 8913,
        "product_count": 77,
        "recommended": false,
        "registration_date": "2018-11-23",
        "review_count": 33,
        "review_date": "2019-01-21",
        "steam_level": 28,
        "updatedAt": "2019-05-20T01:30:23.000Z",
        "user_avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/blakesimkins/128.jpg",
        "user_id": 88435,
        "username": "Salvador",
      }
    }
    source="standard"
  />
</styled.div>
`;
